content/A prediction experiment with Machine Learning - Bruno Michetti.md

```diff
@@ -0,0 +1,153 @@
+
+### Metrics and results
+
+How do we know if the regression model works well? We can use some metrics to analyze that. As I mentioned, it’s a common practice to split the dataset into training and testing datasets. The training dataset it’s used by the model to learn, and the testing dataset it’s used to evaluate the correctness of the resultant model.
+How do we do this evaluation? By the moment of the evaluation, we have a trained model, and a testing dataset that has two main parts: a set of instances of the data, let’s say X_test, and a set of the corresponding correct result for each instance in X_test, let’s call it Y_test. A good way to measure how well our model works is by applying the predict function on X_test, and comparing that result with the correct one (Y_test). To make the comparison we have a couple of metrics that we can use:
+
+<img src="images/prediction-error-definitions.png" alt="Errors" width="800" />
+
+As you can figure out, the closer to 0, the better results.
+In the experiment I used R-squared, Mean squared, and Root mean squared error. For each trained model, I did a measurement of those errors, and also, I used the created scalers for the target variable to scale back the result and compare the values in the original range. Here I show you can see the results of each model:
+
+Results with scaled data
+| Model | R-squared error | Mean squared error | Root mean squared error |
+|---------|-----------------------------|-----------------|-----------------------------------------------|
+| Lasso | 0.57123| 0.01519 | 0.12324 |
+| Ridge | 0.94795 | 0.00184 | 0.04294 |
+| Random Forest | 0.94275 | 0.00202 | 0.04503 |
+
+Results in the original range
+| Model | Mean squared error | Root mean squared error |
+|---------|-----------------|-----------------------------------------------|
+| Lasso | 37.9725| 6.16219 |
+| Ridge | 4.60964 | 2.14701 |
+| Random Forest | 5.06985 | 2.25163 |
+
+The conclusion is that Ridge and Random Forest regression work better than Lasso for this dataset, given that the errors are closer to 0. Besides, scaling back and comparing in the original range, we saw that the error in those cases are both approximately 2 years, a good threshold to predict life expectancy.
+
+### Summary
+
+In this blog I talked about a prediction experiment with machine learning. It was very interesting to me and I learned a lot. I hope that at this point you enjoyed your reading, and understood the main ideas of the general process of prediction. Also, I mentioned important concepts of machine learning such as supervised learning, preprocessing the data, training and testing the models, etc. We saw some of the most used models in regression, and a way to evaluate them in order to see how well they are working. Finally, after reading this blog I hope that you can feel inspired to work with machine learning because it’s fun and helps us to solve lots of amazing and interesting problems.
```

Comment:
I would change "I hope that you can feel inspired" to "I hope you feel inspired" and "helps us to solve" to "helps us solve". WDYT?